src/walletRPC.php

```diff
@@ -425,16 +437,24 @@ public function getheight()
    */
   public function transfer($amount, $address = '', $payment_id = '', $mixin = 6, $account_index = 0, $subaddr_indices = '', $priority = 2, $unlock_time = 0, $do_not_relay = false)
   {
-    if (is_array($amount)) { // Parameters passed in as object
+    if (is_array($amount)) { // Parameters passed in as object/dictionary
       $params = $amount;
 
       if (array_key_exists('destinations', $params)) {
         $destinations = $params['destinations'];
 
-        foreach ($destinations as $key => $amount) {
-          if ($key == 'amount') {
-            // Convert from moneroj to tacoshi (piconero)
-            $destinations[$key] = $amount * 1000000000000;
+        if (!is_array($destinations)) {
+          throw new Exception('Error: destinations must be an array');
+        }
+
+        foreach ($destinations as $destination) {
+          if (array_key_exists('amount', $destinations[$destination])) {
+            $destinations[$destination]['amount'] = $destinations[$destination]['amount'] * 1000000000000;
```

Comment:
There are a lot of  $variable * 1000000000000; with the risk you could left a "0" during the development. I think we should have a function in order to "transform" values.

Reply:
wise decision, I'll add that (back) in a few moments

Reply:
done in 39c14dcc8b14de0b01e8bbfc30a21e83a06794c2 below